middleware/error-handler.js

```diff
@@ -1,10 +1,38 @@
-const { CustomAPIError } = require('../errors')
+// const { CustomAPIError } = require('../errors')
 const { StatusCodes } = require('http-status-codes')
+
 const errorHandlerMiddleware = (err, req, res, next) => {
-  if (err instanceof CustomAPIError) {
-    return res.status(err.statusCode).json({ msg: err.message })
+
+  let customError = {
+    //set Default
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || 'Something Went Wrrong, try again later'
   }
-  return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ err })
+  // if (err instanceof CustomAPIError) {
+  //   return res.status(err.statusCode).json({ msg: err.message })
+  // }
+
+  //Validation Error
+  if (err.name === 'ValidationError') {
+    console.log(Object.values(err.errors));
+    customError.msg = Object.values(err.errors).map((item) => item.message).join(' & ')
+    customError.statusCode = 400
+  }
+
+  //Cast Error
+  if (err.name === 'CastError') {
```

Comment:
I like how you called out a bunch of different error types here and labeled. In a full production build you want might to focus on name or code, but just flagging something you’d see in a bigger build